stmhal/boards/stm32f439.ld

```diff
@@ -0,0 +1,126 @@
+/* Specify the memory areas */
+MEMORY
+{                                                   
+    FLASH (rx)      : ORIGIN = 0x08000000, LENGTH = 0x0200000 /* entire flash, 2048 KiB */
+    FLASH_ISR (rx)  : ORIGIN = 0x08000000, LENGTH = 0x0004000 /* sector 0, 16 KiB */
+    FLASH_TEXT (rx) : ORIGIN = 0x08020000, LENGTH = 0x0080000 /* sectors 5,6,7,8, 4*128KiB = 512 KiB (could increase it more) */
+    RAM (xrw)       : ORIGIN = 0x20000000, LENGTH = 0x0030000 /* 192 KiB */
+	CCMRAM (xrw)    : ORIGIN = 0x10000000, LENGTH = 0x010000 /* 64 KiB */
+}
+ 
+/* produce a link error if there is not this amount of RAM for these sections */
+_minimum_stack_size = 2K;
+_minimum_heap_size = 16K;
+ 
+/* top end of the stack */
+
+/*_stack_end = ORIGIN(RAM) + LENGTH(RAM);*/
+_estack = ORIGIN(RAM) + LENGTH(RAM) - 1;
```

Comment:
@dhylands you are right... that's not good!  I think the reason it works is because the MCU automagically aligns the pointer (or ignores the lower 2 bits).  In that case the -1 loses us 4 bytes of RAM.  I've seen this -1 problem in many places, and am pretty sure it's wrong, but would be good to verify it (yet again) and then fix all occurrences in our code.